### PR TITLE
[ADC Build Image] Add support for build image in ADC

### PR DESCRIPTION
**Pull request by hgreebe**
### Description of changes
* Add support for build image in ADC
* Patch AWS_CA_BUNDLE to make awscli work in ADC

### Tests
* Ran createami integration tests
* Tested build-image command in ADC

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-awsbatch/recipes/install.rb [hunk 1]
```diff
@@ -42,8 +45,8 @@
       curl --retry 3 -L -o aws-parallelcluster.tgz ${custom_package_url}
       mkdir aws-parallelcluster-awsbatch-cli
       tar -xzf aws-parallelcluster.tgz --directory aws-parallelcluster-awsbatch-cli
-      cd aws-parallelcluster-awsbatch-cli/*aws-parallelcluster-*
 
+      cd aws-parallelcluster-awsbatch-cli/*aws-parallelcluster-*
```
Comment (hanwen-pcluste) [inline cookbooks/aws-parallelcluster-awsbatch/recipes/install.rb]: Minor: this is not a change, right?

File: cookbooks/aws-parallelcluster-environment/resources/efa/efa_alinux2.rb [hunk 2]
```diff
@@ -24,6 +24,10 @@ def conflicting_packages
   end
 
   def prerequisites
-    %w(environment-modules libibverbs-utils librdmacm-utils)
+    if aws_region.start_with?("us-iso")
+      %w(libibverbs-utils librdmacm-utils)
+    else
+      %w(environment-modules libibverbs-utils librdmacm-utils)
+    end
```
Comment (hanwen-pcluste) [inline cookbooks/aws-parallelcluster-environment/resources/efa/efa_alinux2.rb:31]: Why is environment-modules removed?

File: cookbooks/aws-parallelcluster-platform/recipes/install/cuda.rb [hunk 3]
```diff
@@ -26,8 +26,8 @@
 cuda_arch = arm_instance? ? 'linux_sbsa' : 'linux'
 cuda_url = "#{node['cluster']['artifacts_s3_url']}/dependencies/cuda/cuda_#{cuda_complete_version}_#{cuda_version_suffix}_#{cuda_arch}.run"
 cuda_samples_version = '12.2'
-cuda_samples_url = "#{node['cluster']['artifacts_s3_url']}/dependencies/cuda/samples/v#{cuda_samples_version}.tar.gz"
 tmp_cuda_run = '/tmp/cuda.run'
+cuda_samples_url = "#{node['cluster']['artifacts_s3_url']}/dependencies/cuda/samples/v#{cuda_samples_version}.tar.gz"
```
Comment (hanwen-pcluste) [inline cookbooks/aws-parallelcluster-platform/recipes/install/cuda.rb]: Minor: this is not a change, right?

File: cookbooks/aws-parallelcluster-platform/resources/gdrcopy/gdrcopy_amazon2.rb [hunk 4]
```diff
@@ -25,6 +25,14 @@ def gdrcopy_platform
   'amzn-2'
 end
 
+def gdrcopy_build_dependencies
+  if aws_region.start_with?("us-iso")
+    %w(dkms rpm-build make check check-devel)
+  else
+    %w(dkms rpm-build make check check-devel subunit subunit-devel)
```
Comment (hanwen-pcluste) [inline cookbooks/aws-parallelcluster-platform/resources/gdrcopy/gdrcopy_amazon2.rb:32]: Why some is removed?

File: cookbooks/aws-parallelcluster-platform/recipes/install/cuda.rb (hunk 3 above)
Comment (hgreebe) [reply cookbooks/aws-parallelcluster-platform/recipes/install/cuda.rb]: correct

File: cookbooks/aws-parallelcluster-platform/resources/gdrcopy/gdrcopy_amazon2.rb (hunk 4 above)
Comment (hgreebe) [reply cookbooks/aws-parallelcluster-platform/resources/gdrcopy/gdrcopy_amazon2.rb:32]: those are not available in ADC regions, so they are downloaded in install_packages through the `epel_deps.tar.gz` artifiact.

File: cookbooks/aws-parallelcluster-awsbatch/recipes/install.rb (hunk 1 above)
Comment (hgreebe) [reply cookbooks/aws-parallelcluster-awsbatch/recipes/install.rb]: yes

File: cookbooks/aws-parallelcluster-environment/resources/efa/efa_alinux2.rb (hunk 2 above)
Comment (hgreebe) [reply cookbooks/aws-parallelcluster-environment/resources/efa/efa_alinux2.rb:31]: same reasoning as the other comment I left

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**